### PR TITLE
Cleanup deployment.yaml

### PR DESCRIPTION
**Pull request by yuzhouliu9**
During kanopy module work I found a few things that I think should be changed to better align with the actual deployment.

### CODE REVIEW

File: examples/k8s/deployment.yaml [hunk 1]
```diff
@@ -113,10 +116,10 @@ webhooks:
       port: 8443
       namespace: "devops"
   sideEffects: None
-  admissionReviewVersions: ["v1beta1"]
+  admissionReviewVersions: ["v1", "v1beta1"]
   failurePolicy: Ignore
-  name: v1beta1.kanopy-platform.github.io
-  reinvocationPolicy: IfNeeded
+  name: hedgetrimmer.kanopy-platform.github.io
```
Comment (yuzhouliu9) [inline examples/k8s/deployment.yaml]: I wasn't sure why for the gateway-certificate-controller we had this as v1beta1, but I think it should deserve a proper name?

File: examples/k8s/deployment.yaml [hunk 2]
```diff
@@ -113,10 +116,10 @@ webhooks:
...
+  reinvocationPolicy: Never
```
Comment (yuzhouliu9) [inline examples/k8s/deployment.yaml]: During initial setup I copied pasted `IfNeeded` here. Looking at [ReinvocationPolicy](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#reinvocation-policy), `Never` should be ok for us and is the default. I tested this setting for a few resources and it works ok.

File: examples/k8s/deployment.yaml [hunk 3]
```diff
@@ -113,10 +116,10 @@ webhooks:
       port: 8443
       namespace: "devops"
   sideEffects: None
-  admissionReviewVersions: ["v1beta1"]
+  admissionReviewVersions: ["v1", "v1beta1"]
```
Comment (yuzhouliu9) [inline examples/k8s/deployment.yaml:119]: Adding `v1` [AdmissionReviewVersion](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#webhook-request-and-response) for completeness

File: examples/k8s/deployment.yaml (hunk 2 above)
Comment (whwalter) [reply examples/k8s/deployment.yaml]: We need reinvocation for things like sidecar injection.

Comment (yuzhouliu9) [reply examples/k8s/deployment.yaml]: Oh, would a possible flow be like we get it first, mutate. Then gets to istio, which injects sidecar, but it'll need to come back to us to add the resources requirements to the sidecar?

Comment (whwalter) [reply examples/k8s/deployment.yaml]: Yeah. hedgetrimmer cares about all the container and initcontainer definitions and the sidecar mutatingwebhook would inject another container. That only happens at the pod level but over all we care if something else changes the list of containers or initcontainers on any of of the workload types.

Comment (yuzhouliu9) [reply examples/k8s/deployment.yaml]: Gotcha, reverted to `IfNeeded` and will change the kanopy module to match

File: examples/k8s/deployment.yaml (hunk 1 above)
Comment (dskatz) [reply examples/k8s/deployment.yaml]: `v1beta1.hedgetrimmer.kanopy-platform.github.io`

**Review by dskatz — APPROVED** [review]

**Pull request merged**